src/main/java/org/gridsuite/filter/server/AbstractFilterRepositoryProxy.java

```diff
@@ -7,33 +7,22 @@
 
 package org.gridsuite.filter.server;
 
-import org.gridsuite.filter.server.dto.AbstractFilter;
-import org.gridsuite.filter.server.dto.AbstractGenericFilter;
-import org.gridsuite.filter.server.dto.AbstractInjectionFilter;
-import org.gridsuite.filter.server.dto.FilterAttributes;
-import org.gridsuite.filter.server.dto.NumericalFilter;
-import org.gridsuite.filter.server.entities.AbstractFilterEntity;
-import org.gridsuite.filter.server.entities.AbstractGenericFilterEntity;
-import org.gridsuite.filter.server.entities.AbstractInjectionFilterEntity;
-import org.gridsuite.filter.server.entities.NumericFilterEntity;
+import com.powsybl.commons.PowsyblException;
+import org.gridsuite.filter.server.dto.*;
+import org.gridsuite.filter.server.entities.*;
 import org.gridsuite.filter.server.repositories.FilterMetadata;
 import org.gridsuite.filter.server.repositories.FilterRepository;
 import org.gridsuite.filter.server.utils.FilterType;
 
-import java.util.Date;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Optional;
-import java.util.Set;
-import java.util.UUID;
+import java.util.*;
 import java.util.stream.Stream;
 
 /**
  * @author Jacques Borsenberger <jacques.borsenberger at rte-france.com>
  * @author Franck Lecuyer <franck.lecuyer at rte-france.com>
  */
 
-abstract class AbstractFilterRepositoryProxy<FilterEntity extends AbstractFilterEntity, EntityRepository extends FilterRepository<FilterEntity>> {
+public abstract class AbstractFilterRepositoryProxy<F extends AbstractFilterEntity, R extends FilterRepository<F>> {
```

Comment:
not very fan of 1 letter Definition, but it is a sonar smell... so i won't complain about it

Reply:
Yes I changed for sonar